plugins/groovy/groovy-psi/src/org/jetbrains/plugins/groovy/codeInspection/changeToOperator/ChangeToOperatorInspection.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2000-2015 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jetbrains.plugins.groovy.codeInspection.changeToOperator;
+
+import com.intellij.codeInspection.LocalQuickFix;
+import com.intellij.codeInspection.ProblemDescriptor;
+import com.intellij.codeInspection.ui.MultipleCheckboxOptionsPanel;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.PsiModifier;
+import com.intellij.psi.PsiModifierListOwner;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.plugins.groovy.codeInspection.BaseInspection;
+import org.jetbrains.plugins.groovy.codeInspection.BaseInspectionVisitor;
+import org.jetbrains.plugins.groovy.codeInspection.GroovyFix;
+import org.jetbrains.plugins.groovy.codeInspection.changeToOperator.data.OptionsData;
+import org.jetbrains.plugins.groovy.codeInspection.changeToOperator.data.ReplacementData;
+import org.jetbrains.plugins.groovy.codeInspection.changeToOperator.transformations.Transformation;
+import org.jetbrains.plugins.groovy.codeInspection.changeToOperator.transformations.Transformations;
+import org.jetbrains.plugins.groovy.lang.psi.api.statements.expressions.path.GrMethodCallExpression;
+
+import javax.swing.*;
+import java.util.Map;
+
+import static com.intellij.codeInspection.ProblemHighlightType.WEAK_WARNING;
+import static org.jetbrains.plugins.groovy.codeInspection.GroovyInspectionBundle.message;
+
+public class ChangeToOperatorInspection extends BaseInspection {
+  private static final Logger LOG = Logger.getInstance(Transformation.class);
+  private static final Map<String, Transformation> TRANSFORMATIONS = Transformations.get();
+
+  public boolean useDoubleNegation = true;
+  public boolean shouldChangeCompareToEqualityToEquals = true;
+
+  @NotNull
+  @Override
+  protected BaseInspectionVisitor buildVisitor() {
+    return new BaseInspectionVisitor() {
+      @Override
+      public void visitMethodCallExpression(GrMethodCallExpression methodCallExpression) {
+        super.visitMethodCallExpression(methodCallExpression);
+        processMethodCall(methodCallExpression);
+      }
+
+      private void processMethodCall(GrMethodCallExpression methodCallExpression) {
+        try {
+          PsiMethod method = methodCallExpression.resolveMethod();
```

Comment:
`reolveMethod` isn't working for complex scenarios, though (e.g. `[1,2,3].isCase(0)`)